### PR TITLE
doc/types-chart

### PR DESCRIPTION
**Pull request by sophiebremer**
TypeScript-compatible doclets

### CODE REVIEW

File: js/parts/Chart.js [hunk 1]
```diff
@@ -1242,24 +1361,34 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
      * Resize the chart to a given width and height. In order to set the width
      * only, the height argument may be skipped. To set the height only, pass
      * `undefined` for the width.
-     * @param  {Number|undefined|null} [width]
+     *
+     * @function Highcharts.Chart#setSize
+     *
+     * @param  {number|null} [width]
```
Comment (pawelfus) [inline js/parts/Chart.js]: We don't define `undefiend` anymore?

Comment (sophiebremer) [reply js/parts/Chart.js]: In TypeScript every optional parameter implies support of `undefined`. So if you like to skip an optional parameter to get to an other optional parameter, you are always allowed to set the one to skip to `undefined`.

Comment (sophiebremer) [reply js/parts/Chart.js]: Do you think, we should add to every optional parameter the `undefined` type, to make this clear?

Comment (pawelfus) [reply js/parts/Chart.js]: I'm just worried it might be confusing for users. Like "is undefined deprecated?"

Comment (sophiebremer) [reply js/parts/Chart.js]: I see your point. I will add `undefined` to optionals, to make it clear for everyone.

File: js/parts/Chart.js [hunk 2]
```diff
@@ -4,6 +4,27 @@
  * License: www.highcharts.com/license
  */
 
+/**
+ * The chart title. The title has an `update` method that allows modifying the
+ * options directly or indirectly via `chart.update`.
+ *
+ * @typedef {Highcharts.SVGElement} Highcharts.TitleObject
+ *
+ * @property {Function} update
```
Comment (TorsteinHonsi) [inline js/parts/Chart.js]: The parameters are the same as `TitleOptions`.

File: js/parts/Chart.js [hunk 3]
```diff
@@ -4,6 +4,27 @@
...
+ */
+
+/**
+ * The chart subtitle. The subtitle has an `update` method that
+ * allows modifying the options directly or indirectly via
+ * `chart.update`.
+ *
+ * @typedef {Highcharts.SVGElement} Highcharts.SubtitleObject
+ *
+ * @property {Function} update
```
Comment (TorsteinHonsi) [inline js/parts/Chart.js]: The parameters are the same as `SubtitleOptions`.

**Pull request merged**